### PR TITLE
Use verified doubles in specs

### PR DESCRIPTION
**Pull request by MGotink**
As noted by @bf4 in #258.

### CODE REVIEW

File: spec/metric_fu/metrics/reek/generator_spec.rb [hunk 1]
```diff
@@ -48,46 +51,53 @@
...
+
+            instance_double(Reek::Smells::SmellWarning,
+                            source: "app/controllers/newline_controller.rb",
+                            context: "NewlineController#some_method",
+                            message: "calls current_user.<< \"new line\n\" multiple times",
+                            smell_type: "Duplication",
+                            lines: [6, 9])
         ]
-        @lines = double(smells: @smells)
+        @lines = instance_double(Reek::Core::Examiner, smells: @smells)
```
Comment (bf4) [inline spec/metric_fu/metrics/reek/generator_spec.rb]: so, this test will fail on Reek versions before the Reek::Examiner was renamed to Reek::Core::Examiner, but I'm ok with it for now.

File: spec/metric_fu/metrics/reek/generator_spec.rb [hunk 2]
```diff
@@ -39,6 +39,9 @@
   end
 
   describe "analyze method" do
+    require "reek"
+    require "reek/cli/application"
```
Comment (bf4) [inline spec/metric_fu/metrics/reek/generator_spec.rb]: might be better to refactor the [`run!` method in the generator](https://github.com/metricfu/metric_fu/blob/115c5f1a0d906af49529c0b67846e71127d9f808/lib/metric_fu/metrics/reek/generator.rb#L18-L24) so we can use this logic just the once to get the examiner and small classes...

**Comment by bf4** [comment]
Is missing config in spec_helper

```diff
config.mock_with :rspec do |mocks|
    mocks.syntax = :expect
+    mocks.verify_partial_doubles = true
 end
```

**Comment by MGotink** [comment]
Hmm, 1.9.2 is failing, no idea why, do we still want to support it?

**Comment by bf4** [comment]
It's probably evaling the const before it's defined by the require statement... so, either we can refactor now to use the code from the generator, or just do the release without this then take our time on the refactor

**Comment by bf4** [comment]
Since you're doing the work right now, you can decide :)

File: spec/metric_fu/metrics/reek/generator_spec.rb [hunk 3]
```diff
@@ -43,51 +43,63 @@
       MetricFu::Configuration.run {}
       allow(File).to receive(:directory?).and_return(true)
       @reek = MetricFu::ReekGenerator.new
+      @examiner = @reek.send(:examiner)
+      @smell_warning = Reek.const_defined?(:SmellWarning) ? Reek.const_get(:SmellWarning) : Reek.const_get(:Smells).const_get(:SmellWarning)
+      if @smell_warning.instance_methods.include?(:subclass)
```
Comment (MGotink) [inline spec/metric_fu/metrics/reek/generator_spec.rb]: I'm not too happy with this kind of hacks in the specs, but I can't think of a better solution right now.
Ruby 1.9.2. is using an older reek version which doesn't implement SmellWarning#smell_type, due to which the verified double fails.
OTOH the older reek output is tested separately.

@bf4, maybe you have a better solution?

**Comment by MGotink** [comment]
@bf4, any ideas for the old reek fix, or just keep it this way for now?

**Comment by bf4** [comment]
I suppose this way for now...

B mobile phone

**Comment by bf4** [comment]
I just pushed 4.12.0. I couldn't remember my password, so didn't sign it.. will bump a patch if I can remember.

**Pull request merged**